mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -959,21 +1023,75 @@ func expandAutoHealingPolicies(configured []interface{}) []*compute.InstanceGrou
 	return autoHealingPolicies
 }
 
-func expandStatefulPolicy(configured []interface{}) *compute.StatefulPolicy {
+func expandStatefulPolicy(d *schema.ResourceData) *compute.StatefulPolicy {
+	preservedState := &compute.StatefulPolicyPreservedState{
+	}
+	stateful_disks := d.Get("stateful_disk").(*schema.Set).List()
 	disks := make(map[string]compute.StatefulPolicyPreservedStateDiskDevice)
-	for _, raw := range configured {
+	for _, raw := range stateful_disks {
 		data := raw.(map[string]interface{})
 		disk := compute.StatefulPolicyPreservedStateDiskDevice{
 			AutoDelete: data["delete_rule"].(string),
 		}
 		disks[data["device_name"].(string)] = disk
 	}
-	if len(disks) > 0 {
-		return &compute.StatefulPolicy{PreservedState: &compute.StatefulPolicyPreservedState{Disks: disks}}
+	preservedState.Disks = disks
+	var change bool
+	change = false
+	<% unless version == "ga" -%>
+	if d.HasChange("stateful_internal_ip") {
+		oldInternalIps, newInternalIps := d.GetChange("stateful_internal_ip")
+		if len(newInternalIps.([]interface{})) == 0 {
+			preservedState.NullFields = append(preservedState.NullFields, "internalIPs")
+		} else {
+			preservedState.InternalIPs = expandStatefulIps(oldInternalIps.([]interface{}), newInternalIps.([]interface{}), preservedState, "InternalIPs.%s")
```

Comment:
as discussed - we should remove this expander and just use `d.Get` - simpler. If there's a need to suppress a diff we should use a set hash func or diff suppress func rather than trying to handle it here.